File: 724-find-pivot-index/find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        int totalSum = 0;
        for(int num: nums) totalSum += num;
        int rightSum = totalSum;
        int leftSum = 0;
        for(int i=0; i<n; i++) {
            rightSum -= nums[i];
            if(leftSum == rightSum) return i;
            leftSum += nums[i];
        }
        return -1;
    }
};

File: 724-find-pivot-index/find-pivot-index.ts
function pivotIndex(nums: number[]): number {
    const totalSum = nums.reduce((sum, num) => sum + num, 0);
    let leftSum = 0, rightSum = totalSum;
    let ans = -1;

    for(let i=0; i<nums.length; i++) {
        rightSum -= nums[i];
        if(rightSum == leftSum) {
            ans = i;
            break;
        }
        leftSum += nums[i];
    }

    return ans;
};